src/Microsoft.AspNetCore.Authentication.Facebook/FacebookHandler.cs

```diff
@@ -35,7 +36,11 @@ public FacebookHandler(HttpClient httpClient)
             }
 
             var response = await Backchannel.GetAsync(endpoint, Context.RequestAborted);
-            response.EnsureSuccessStatusCode();
+            if (!response.IsSuccessStatusCode)
+            {
+                var errorMessage = $"Failed to retrived Facebook user information ({response.StatusCode}) Please check if the authentication information is correct and the corresponding Google API is enabled.";
```

Comment:
FWIW, here's what we do in all the aspnet-contrib OAuth2 providers:

```csharp
var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
if (!response.IsSuccessStatusCode) {
    Logger.LogError("An error occurred when retrieving the user profile: the remote server " +
                    "returned a {Status} response with the following payload: {Headers} {Body}.",
                    /* Status: */ response.StatusCode,
                    /* Headers: */ response.Headers.ToString(),
                    /* Body: */ await response.Content.ReadAsStringAsync());

    throw new HttpRequestException("An error occurred when retrieving the user profile.");
}
```

Ideally, we shouldn't have to throw an exception, but returning `null` is not legal as `OAuthHandler.HandleRemoteAuthenticateAsync` crashes if the ticket is null.

IMHO, you should fix `HandleRemoteAuthenticateAsync` to detect if the ticket returned by `CreateTicketAsync` is null and ultimately return `AuthenticateResult.Failed` instead of calling `AuthenticateResult.Success` blindly.

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/dev/src/AspNet.Security.OAuth.GitHub/GitHubAuthenticationHandler.cs#L33-L39